<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Let's Promise</title>
</head>
<body>
<script>
    /*
    let myPromise = new Promise((resolve, reject) => {
        let request = new XMLHttpRequest();
        request.open('GET','http://localhost:8080/batman');
        request.onload = () => {
            resolve(request.responseText);
        };
        request.send();
    });
     */
    asyncBatman();
    // syncBatman();
    async function asyncBatman() {
        console.log('Do before');
        try {
            let response = await fetch('http://localhost:8080/batman');
            let responseText = await response.text();
        } catch (reason) {
            console.log(reason);
        }
        console.log(response);
        console.log(responseText);
        console.log('Do after');
    }
    function syncBatman() {
        console.log('Do before in syncBatman');
        fetch('http://localhost:8080/batman', {
            method: "POST"
        }).then((response) => {
            console.log(response);
            response.text().then((responseText) => {
                console.log(responseText);
            });
            return response.text();
        }).then((responseText) => {
            console.log(responseText);
        }).catch((reason) => {
            console.log(reason);
            console.log('Some error happend');
        }).finally(() => {
            console.log('Finally happens this no matter what');
        });
        console.log('Do after in syncBatman');
    }
    /*
    fetchPromise.then((responseText) => {
        setTimeout(() => {
            console.log('done first');
        }, 1000);
        return new Promise((resolve, reject) => {
            resolve('now Finished');
        });
    }).then((previousPromiseResolved) => {
        console.log(previousPromiseResolved);
    }).catch((err) => {
        console.dir(err);
    });
     */
</script>
</body>
</html>